lib/puppet/module/task.rb

```diff
@@ -61,6 +62,37 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    # Find task's required lib files and retrieve paths
+    # for both 'files' and 'implementation:files' metadata keys
+    def self.find_files(files, mod)
+      env = mod.environment.respond_to?(:name) ? mod.environment.name : 'production'
+
+      file_list = files.map do |file|
+        module_name, mount, endpath = file.split("/", 3)
+
+        pup_module = Puppet::Module.find(module_name, env)
+        if pup_module.nil?
+          raise Puppet::Module::MissingModule, _("Module %{module_name} not found in environment %{environment_name}.") %
+            {module_name: pup_module.name, environment_name: env}
+        end
+
+        unless MOUNTS.include? mount
+          msg = _("Files must be saved in module directories that Puppet makes available via mount points: %{mounts}" % 
+                  {mounts: MOUNTS.join(', ')})
+          raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+        end
+
+        path = File.join(pup_module.path, mount, endpath)
+        unless Pathname.new(path).absolute?
```

Comment:
This will always be true, because we're joining it with the path to the module, which is absolute. `/foo/../bar` is an 
absolute path

Reply:
Oh, then we may want one of the other `absolute?` implementations. I think there was one that ensures there are no `..` in the path.

Reply:
The safest option may be to go back to my original proposal at https://github.com/puppetlabs/puppet/pull/7042#discussion_r215768507.